Added option for user to choose the infra and node network interface …

satishoc opened the pull request.

satishoc committed:
Added option for user to choose the infra and node network interface type

snaiksat commented on update_ign.py:
```diff
@@ -38,201 +38,379 @@
+    elif ans == "4":
+        choice = 4
+    else:
+        print("\n   Invalid Option, Please Try Again")
```
Could you please provide information about these options as a part of the comments section we have at the top of this script? We intend that to be the documentation on how to run this script and what options it supports.

satishoc replied:
Added

satishoc committed:
changed variable name for node and infra interface, Added description for the supported option

satishoc committed:
Removed extra spaces

abhis2112 commented on update_ign.py:
```diff
@@ -37,202 +49,365 @@
+        return ifcfg_bond0
+
+
+    def create_interface_with_bond(self, interface_name, bond_name, mtu):
```
create_slave_interface

satishoc replied:
renamed

abhis2112 commented on update_ign.py:
```diff
@@ -37,202 +49,365 @@
+            infra_network_interface2_b64 = base64.standard_b64encode(infra_network_interface2).decode().strip()
+            config_data[interface_name] = {'base64': infra_network_interface2_b64, 'path': '/etc/sysconfig/network-scripts/' + interface_name}
+
+            opflex_conn = ("""VLAN=yes
```
this code is repeated. Please modularize it

abhis2112 commented on update_ign.py:
```diff
@@ -37,202 +49,365 @@
+
+            route_opflex_conn_b64 = base64.standard_b64encode(route_opflex_conn).decode().strip()
+
+            config_data['route_opflex_conn'] = {'base64': route_opflex_conn_b64, 'path': '/etc/sysconfig/network-scripts/route-opflex-conn'}
```
till here. Make a separate function if this code will be called again and again

satishoc replied:
done

abhis2112 commented on update_ign.py:
```diff
@@ -37,202 +49,365 @@
+
+        if 'bootstrap' not in hostname.decode():
+
+            for k, v in config_data.items():
```
make it more verbose. Like - interface, interface_config --> interface_config[path]

satishoc replied:
done

satishoc committed:
refactor code

snaiksat commented on update_ign.py:
```diff
@@ -21,7 +21,19 @@
+#                    4.Single interface for both node and infra networks
+#
+#
+#               currently we support option one only(please update interface name in config.yaml file)
```
> currently we support option one only

This comment is fine in the commit message but does not make sense here.  

> (please update interface name in config.yaml file)

This should be removed from the parentheses and clearly stated what changes need to be made in each case. So say something to the effect of:

    Please update the config.yaml as follows:

snaiksat commented on update_ign.py:
```diff
@@ -21,7 +21,19 @@
+
+
+# How to run this script: 
+#           Please update the config.yaml file before running this script
```
Please remove this comment

snaiksat commented on update_ign.py:
```diff
@@ -21,7 +21,19 @@
+
+# How to run this script: 
+#           Please update the config.yaml file before running this script
+#           Trigger the command after changing config.yaml file according to requirement:   python3 update_ign.py
```
Please revert this back to how it was earlier

snaiksat commented on update_ign.py:
```diff
@@ -21,7 +21,19 @@
 #                     In either of those cases this script will have to be modified to account for those differences.
 
-# How to run this script: python3 update_ign.py
+#               we will have following option to choose from 
```
How does this sentence tie into the existing text comments above it??

satishoc committed:
Revert the changes

satishoc committed:
Removed harcoded value for interface name

satishoc committed:
Removed unwanted comment

snaiksat commented:
Thanks @satishoc for updating the comment. @abhis2112  when you get a chance please check if the generated output provided by @satishoc for all the cases matches the expectations.

abhis2112 commented:
lgtm

camrossi committed:
Update update_ign.py

Remove an extra space after DEVICE for case 4. This will result in the config not to be correctly applied an won't create the opflex interfaxce

PariyaAshok committed:
Merge pull request #4 from camrossi/patch-1

Update update_ign.py
